### PR TITLE
Task/module 7

### PR DESCRIPTION
**Pull request by AB619**

### CODE REVIEW

File: src/store/movieStore.ts [hunk 1]
```diff
@@ -15,6 +15,13 @@ const useMovieStore = defineStore("movieStore", () => {
     setFilterParam,
     searchMovies,
   } = useSearch(initialList.value);
+  
+  const fetchMovies = async () => {
+    const { movies, error, getAllMovies } = useMovies();
+    await getAllMovies();
+    if(error.value) initialList.value = mockMoviesList;
```
Comment (hicugi) [inline src/store/movieStore.ts]: It's quite confusing that your Promise responding with positive data if some error appears. You also should use reject in this case

**Review by hicugi — CHANGES_REQUESTED** [review]
Minor fix should be made for better understanding & expectations

File: src/utils.ts [hunk 2]
```diff
@@ -58,6 +65,24 @@ export function useSearch(initialMovies: IMockMovie[]) {
...
+
+export function useMovies() {
+  const movies: Ref<IMockMovie[]> = ref([]);
+  const error: Ref<string | null> = ref(null);
+
+  const getAllMovies = async () => {
+    try {
+      const endpoint = "/movies.json";
+      const response = await apiClient(endpoint);
+      movies.value = response.movies;
```
Comment (hicugi) [inline src/utils.ts:80]: This is deffenetlly won't work for you if your request fails. Look closer to Promise chain you have created

**Review by hicugi — CHANGES_REQUESTED** [review]

File: src/store/movieStore.ts (hunk 1 above)
Comment (AB619) [reply src/store/movieStore.ts]: fixed

File: src/utils.ts (hunk 2 above)
Comment (AB619) [reply src/utils.ts:80]: fixed

File: src/store/movieStore.ts [hunk 3]
```diff
@@ -15,6 +15,17 @@ const useMovieStore = defineStore("movieStore", () => {
     setFilterParam,
     searchMovies,
   } = useSearch(initialList.value);
+  
+  const fetchMovies = async () => {
+    const { movies, error, getAllMovies } = useMovies();
+    try{
+      await getAllMovies();
+      initialList.value = movies.value;
+    }catch{
```
Comment (hicugi) [inline src/store/movieStore.ts]: Don't forget to use the Prettier

File: src/store/movieStore.ts [hunk 4]
```diff
@@ -15,6 +15,17 @@ const useMovieStore = defineStore("movieStore", () => {
...
+      initialList.value = [];
+      console.log(error.value);
```
Comment (hicugi) [inline src/store/movieStore.ts]: And ESLint - don't leave logs

**Review by hicugi — CHANGES_REQUESTED** [review]
The search isn't working

**Review by hicugi — APPROVED** [review]
A structure of the store a bit of a mess, but overall - requriements met

**Pull request merged**